trieste/ask_tell_optimization.py

```diff
@@ -457,6 +458,7 @@ def ask(self) -> TensorType:
                             logging.scalar(f"query_points/[{i}]", float(query_points[0, i]))
                         else:
                             logging.histogram(f"query_points/[{i}]", query_points[:, i])
+                    logging.histogram("query_points/euclidean_distances", pdist(query_points))
```

Comment:
Probably fine here, but generally better to use a closure for two reasons: it doesn't waste time doing the computation if you disable the log, and it catches any exception that the computation might raise.

```suggestion
                    logging.histogram("query_points/euclidean_distances", lambda: pdist(query_points))
```

Reply:
done